src/matching/Generic_vs_generic.ml

```diff
@@ -2800,6 +2800,20 @@ and m_list__m_type_parameter a b =
 
 and m_definition_kind a b =
   match (a, b) with
+  (* We maintain an equivalence between a FieldDefColon of an ellipsis
+     with any other definition. This is because FieldDefColon appears
+     when describing the RHS of a record, so this describes a record
+     like
+     { x = ... }
+     The FieldDefColon is only if it's not some other kind of
+     `definition_kind` structure, however, like a `FuncDef`, so we
+     would like to be able to match those with the ellipsis anyways.
+
+     A concrete example is TS, when you have
+     { func: function (opts) { return "whatever"; } }
+     which is a record containing an entry which is a `FuncDef`.
+  *)
+  | G.FieldDefColon { vinit = Some { e = Ellipsis _; _ }; _ }, _ -> return ()
```

Comment:
I would feel more comfortable enumerating the options for B here, in case we add some new construct that shouldn't match. If you're confident that could never happen, making it instead `_any_other_kind_of_def` would be helpful to show that you've thought about it.